racy test: pkg/integration: TestDockerCmdInDirWithTimeout, TestDockerCmdWithTimeout, TestRunCommandWithOutputForDurationKilled, TestRunCommandWithOutputAndTimeoutKilled

**Steps to reproduce the issue:**
Execute `BUILDFLAGS=-race make test-unit`
(Tested on 29dbcbad878483d9239d6a432c85620aced895c4)

**Describe the results you received:**
```
==================
WARNING: DATA RACE
Read by goroutine 21:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xa94
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 27:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:72 +0xb5
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 21 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 27 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================
==================
WARNING: DATA RACE
Read by goroutine 21:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xaab
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 27:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:74 +0xef
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 21 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 27 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================
==================
WARNING: DATA RACE
Read by goroutine 21:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 29:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:110 +0x0
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:169 +0x27f
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 21 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 29 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdInDirWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:87 +0x128
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdInDirWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:319 +0xc8f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================

----------------------------------------------------------------------
FAIL: dockerCmd_utils_test.go:270: DockerCmdSuite.TestDockerCmdInDirWithTimeout

dockerCmd_utils_test.go:326:
    c.Assert(error, check.IsNil, check.Commentf("Expected no error, got %v", error))
... value *errors.errorString = &errors.errorString{s:"\"dir:/tmp/test-docker-cmd-in-dir353438801 run -ti ubuntu echo hello\" failed with errors: command timed out : \"hello\""} ("\"dir:/tmp/test-docker-cmd-in-dir353438801 run -ti ubuntu echo hello\" failed with errors: command timed out : \"hello\"")
... Expected no error, got "dir:/tmp/test-docker-cmd-in-dir353438801 run -ti ubuntu echo hello" failed with errors: command timed out : "hello"

==================
WARNING: DATA RACE
Read by goroutine 59:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xa94
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 65:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:72 +0xb5
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 59 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 65 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================
==================
WARNING: DATA RACE
Read by goroutine 59:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xaab
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 65:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:74 +0xef
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 59 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 65 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================
==================
WARNING: DATA RACE
Read by goroutine 59:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80

Previous write by goroutine 67:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:110 +0x0
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:169 +0x27f
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 59 (running) created at:
  github.com/go-check/check.(*suiteRunner).forkCall()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:670 +0x486
  github.com/go-check/check.(*suiteRunner).forkTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:815 +0x126
  github.com/go-check/check.(*suiteRunner).runTest()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:824 +0xba
  github.com/go-check/check.(*suiteRunner).run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:618 +0x481
  github.com/go-check/check.Run()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x50
  github.com/go-check/check.RunAll()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0x133
  github.com/go-check/check.TestingT()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x696
  github.com/docker/docker/pkg/integration.Test()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:20 +0x2e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Goroutine 67 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.DockerCmdWithTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/dockerCmd_utils.go:57 +0xe6
  github.com/docker/docker/pkg/integration.(*DockerCmdSuite).TestDockerCmdWithTimeout()
      /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils_test.go:205 +0x689
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:437 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:300 +0xcd
  github.com/go-check/check.(*suiteRunner).forkTest.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:783 +0x5e0
  github.com/go-check/check.(*suiteRunner).forkCall.func1()
      /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:669 +0x80
==================

----------------------------------------------------------------------
FAIL: dockerCmd_utils_test.go:162: DockerCmdSuite.TestDockerCmdWithTimeout

dockerCmd_utils_test.go:212:
    c.Assert(error, check.IsNil, check.Commentf("Expected no error, got %v", error))
... value *errors.errorString = &errors.errorString{s:"\"run -ti ubuntu echo hello\" failed with errors: command timed out : \"hello\""} ("\"run -ti ubuntu echo hello\" failed with errors: command timed out : \"hello\"")
... Expected no error, got "run -ti ubuntu echo hello" failed with errors: command timed out : "hello"

OOPS: 6 passed, 2 FAILED
--- FAIL: Test (4.62s)
==================
WARNING: DATA RACE
Read by goroutine 23:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xa94
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 24:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:72 +0xb5
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 23 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 24 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 23:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xaab
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 24:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:74 +0xef
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 23 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 24 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 23:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 24:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:110 +0x0
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:169 +0x27f
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 23 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 24 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 23:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 24:
  syscall.raceWriteRange()
      /usr/local/go/src/syscall/race.go:29 +0x38
  syscall.Read()
      /usr/local/go/src/syscall/syscall_unix.go:163 +0x9d
  os.(*File).read()
      /usr/local/go/src/os/file_unix.go:211 +0x89
  os.(*File).Read()
      /usr/local/go/src/os/file.go:95 +0xbc
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:173 +0x414
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 23 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 24 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputForDurationKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:162 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 27:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xa94
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 31:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:72 +0xb5
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 27 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 31 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 27:
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:166 +0xaab
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 31:
  bytes.(*Buffer).Truncate()
      /usr/local/go/src/bytes/buffer.go:74 +0xef
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:158 +0xcd
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 27 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 31 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 27:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 31:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:110 +0x0
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:169 +0x27f
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 27 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 31 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Read by goroutine 27:
  runtime.slicebytetostring()
      /usr/local/go/src/runtime/string.go:75 +0x0
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 31:
  syscall.raceWriteRange()
      /usr/local/go/src/syscall/race.go:29 +0x38
  syscall.Read()
      /usr/local/go/src/syscall/syscall_unix.go:163 +0x9d
  os.(*File).read()
      /usr/local/go/src/os/file_unix.go:211 +0x89
  os.(*File).Read()
      /usr/local/go/src/os/file.go:95 +0xbc
  bytes.(*Buffer).ReadFrom()
      /usr/local/go/src/bytes/buffer.go:173 +0x414
  io.copyBuffer()
      /usr/local/go/src/io/io.go:375 +0x1a5
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  os/exec.(*Cmd).writerDescriptor.func1()
      /usr/local/go/src/os/exec/exec.go:232 +0x98
  os/exec.(*Cmd).Start.func1()
      /usr/local/go/src/os/exec/exec.go:340 +0x2a

Goroutine 27 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 31 (running) created at:
  os/exec.(*Cmd).Start()
      /usr/local/go/src/os/exec/exec.go:341 +0xe4e
  github.com/docker/docker/pkg/integration.RunCommandWithOutputForDuration()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:126 +0x4c1
  github.com/docker/docker/pkg/integration.RunCommandWithOutputAndTimeout()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:178 +0x74
  github.com/docker/docker/pkg/integration.TestRunCommandWithOutputAndTimeoutKilled()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:202 +0xf5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Write by goroutine 49:
  runtime.closechan()
      /usr/local/go/src/runtime/chan.go:295 +0x0
  github.com/docker/docker/pkg/integration.(*ChannelBuffer).Close()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:415 +0x6a
  github.com/docker/docker/pkg/integration.TestChannelBufferTimeout()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:527 +0x20c
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc

Previous read by goroutine 48:
  runtime.chansend()
      /usr/local/go/src/runtime/chan.go:107 +0x0
  github.com/docker/docker/pkg/integration.(*ChannelBuffer).Write()
      github.com/docker/docker/pkg/integration/_test/_obj_test/utils.go:408 +0xa2
  io.WriteString()
      /usr/local/go/src/io/io.go:282 +0x167
  strings.(*Reader).WriteTo()
      /usr/local/go/src/strings/reader.go:136 +0x136
  io.copyBuffer()
      /usr/local/go/src/io/io.go:371 +0xe0
  io.Copy()
      /usr/local/go/src/io/io.go:351 +0x78
  github.com/docker/docker/pkg/integration.TestChannelBufferTimeout.func1()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:515 +0x13f

Goroutine 49 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/docker/docker/pkg/integration/_test/_testmain.go:158 +0x384

Goroutine 48 (finished) created at:
  github.com/docker/docker/pkg/integration.TestChannelBufferTimeout()
      /go/src/github.com/docker/docker/pkg/integration/utils_test.go:516 +0x126
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:456 +0xdc
==================
FAIL
coverage: 89.1% of statements
FAIL	github.com/docker/docker/pkg/integration	10.970s
```

**Describe the results you expected:**
There should not be any race.

----

I'll open PR to add `-race` to `go test` after https://github.com/docker/docker/pull/22930 gets merged, which simplifies the flow for `go test`.

--- a/pkg/integration/cmd/command.go
+++ b/pkg/integration/cmd/command.go
@@ -8,6 +8,7 @@ import (
 	"path/filepath"
 	"runtime"
 	"strings"
+	"sync"
 	"syscall"
 	"time"
 
@@ -49,6 +50,23 @@ func ProcessExitCode(err error) (exitCode int) {
 	return
 }
 
+type lockedBuffer struct {
+	m   sync.RWMutex
+	buf bytes.Buffer
+}
+
+func (buf *lockedBuffer) Write(b []byte) (int, error) {
+	buf.m.Lock()
+	defer buf.m.Unlock()
+	return buf.buf.Write(b)
+}
+
+func (buf *lockedBuffer) String() string {
+	buf.m.RLock()
+	defer buf.m.RUnlock()
+	return buf.buf.String()
+}
+
 // Result stores the result of running a command
 type Result struct {
 	Cmd      *exec.Cmd
@@ -56,8 +74,8 @@ type Result struct {
 	Error    error
 	// Timeout is true if the command was killed because it ran for too long
 	Timeout   bool
-	outBuffer *bytes.Buffer
-	errBuffer *bytes.Buffer
+	outBuffer *lockedBuffer
+	errBuffer *lockedBuffer
 }
 
 // Assert compares the Result against the Expected struct, and fails the test if
@@ -255,8 +273,8 @@ func buildCmd(cmd Cmd) *Result {
 	default:
 		execCmd = exec.Command(cmd.Command[0], cmd.Command[1:]...)
 	}
-	outBuffer := new(bytes.Buffer)
-	errBuffer := new(bytes.Buffer)
+	outBuffer := new(lockedBuffer)
+	errBuffer := new(lockedBuffer)
 
 	execCmd.Stdin = cmd.Stdin
 	execCmd.Dir = cmd.Dir

